packages/studio/src/components/TailwindPropInput.tsx

```diff
@@ -0,0 +1,181 @@
+import { useCallback, useEffect, useRef, useState } from "react";
+import generateTailwindSafelist from "../utils/generateTailwindSafelist";
+import { StudioTailwindTheme } from "@yext/studio-plugin";
+import { ReactComponent as EmbedIcon } from "../icons/embed.svg";
+import { ReactComponent as X } from "../icons/x.svg";
+import { useRootClose } from "@restart/ui";
+import classNames from "classnames";
+
+const safelistPromise: Promise<string[] | undefined> = import(
+  "@pathToUserProjectRoot/tailwind.config"
+).then((module) => {
+  const theme: StudioTailwindTheme | undefined = module.default?.theme?.extend;
+  return theme && generateTailwindSafelist(theme);
+});
+
+interface Props {
+  onChange: (value: string) => void;
+  value: string;
+  disabled: boolean;
+}
+
+export default function TailwindPropInput({
+  onChange,
+  value,
+  disabled,
+}: Props) {
+  const safelist = useAwaitedSafelist(value);
```

Comment:
Where is it mutating safelist? Do you mean the .filter? I think only immutable methods are being used.
The point of the filter is to prevent the user from adding duplicates, so a `canAddClasses` or similar boolean wouldn't work

Reply:
Yes updated!